tests/drivers/counter/counter_basic_api/src/test_counter.c

```diff
@@ -722,6 +723,10 @@ static void test_valid_function_without_alarm(const struct device *dev)
 		ticks_expected = counter_get_top_value(dev) - ticks_expected;
 	}
 
+	/* counter might not start from 0, use current value as offset */
+	counter_get_value(dev, &tick_current);
+	ticks_expected += tick_current;
+
 	err = counter_start(dev);
```

Comment:
> will fail if the timer resets upon counter_start() call

I think the ordering could be changed to address this 
```suggestion
	err = counter_start(dev);
	/* counter might not start from 0, use current value as offset */
	counter_get_value(dev, &tick_current);
	ticks_expected += tick_current;
```
The new problem I've added is that there will be a number of ticks incremented after starting (so the upper bound for elapsed ticks might need to go up).

I'm happy with either way.

Reply:
This approach makes more sense. It will indeed solve the case where timer is reset. Will update.